dialogue-core/src/main/java/com/palantir/dialogue/core/Responses.java

```diff
@@ -70,20 +72,20 @@ static boolean isRetryableQos(Response result) {
                 || !RetryHint.DO_NOT_RETRY.equals(reason.retryHint().get());
     }
 
-    static boolean isServerErrorRange(Response response) {
-        return response.code() / 100 == 5;
+    static boolean isServerErrorRange(@Nullable Response response) {
+        return response == null || response.code() / 100 == 5;
     }
 
-    static boolean isInternalServerError(Response response) {
-        return response.code() == 500;
+    static boolean isInternalServerError(@Nullable Response response) {
+        return response == null || response.code() == 500;
```

Comment:
if we do not get a response, we now consider that an internal error, where previously this could have triggered a `NullPointerException`, and NPE would trigger internal error.